### ISSUE TITLE
Specified network bridge doesn't get used

### ISSUE THREAD

**Issue by apigban** [inoxio/ansible-role-proxmox-vms#1]
I modified the defaults/main.yml to set the network to a vswitch with internet connection.

Here's what I added:
`  net: '{"net0":"virtio,bridge=vmbr2"}'`
[FULL main.yml](https://pastebin.com/raw/My6UHxzK)

When I executed the sample playbook, the installation keeps getting stuck and fails on **BAD ARCHIVE MIRROR**.

**What I tried:**
Going to to the proxmox UI, the VMs had  this as the bridge:
![image](https://user-images.githubusercontent.com/2720851/57696204-456a2400-7661-11e9-8792-af5a95463a4c.png)

To make it work (create VMs and continue ubuntu installation), I had to re-select the correct (vmbr2) switch from the proxmox GUI.

**Comment by MichaelKunze** [comment]
Which task is failing and could you please attach a relevant execution log?

Thanks!

**Comment by apigban** [comment]
Hi Thanks for the response, I'm quite new to proxmox automation.

**The [ansible-playbook](https://pastebin.com/raw/4GDeMPDC)** 

**The ansible-playbook invocation I used:** 
`ansible-playbook -i hosts playbooks/create_Rancher.yaml --ask-become-pass`

**The task that failed:** 
`TASK [inoxio.proxmox_vms : Wait for virtual machines to finish installation] ****************************************`

The execution log is quite hard to sanitize, this is the part where it fails [execution log](https://pastebin.com/raw/bcmAZvMd).

when you view the proxmox console for the vm:

![image](https://user-images.githubusercontent.com/2720851/57756439-4d2ed480-7704-11e9-8407-e49879e29ee2.png)

Basically, the vm cannot communicate with the network because it is using vmbr0, instead of the assigned vmbr2 (default vmbr0 was changed t0 vmbr2 in defaults/main.yml)

**Comment by apigban** [comment]
Nevermind, I got it working. Apparently I need to specify the network bridge and preseed template in the playbook, like this:


 ```
- hosts: REDACTED # see hosts file
  remote_user: REDACTED
  become: true
 roles:
    - role: inoxio.proxmox_vms
      proxmox:
        api_user: REDACTED
        api_password: REDACTED
        api_host: REDACTED
      vms:
        REDACTED:
          node: 'proxmox'
          locale: en_US
          ubuntu_distribution: 'bionic'
          root_password:  'REDACTED'
          **net: '{"net0":"virtio,bridge=vmbr2"}'**
          **preseed_template: 'ubuntu-bionic.seed'**
          memory_size: 8192
          virtio: '{"virtio0":"local:30,cache=writeback,discard=on"}'
          network:
            ip: REDACTED
            netmask: REDACTED
            gateway: REDACTED
            nameserver: REDACTED 1.1.1.1 8.8.8.8
            domainname: REDACTED
          additional_packages:
             - curl
             - gnupg
```

**Comment by MichaelKunze** [comment]
Thank you for leaving the solution.
On a side note: The solution is also the proposed and documented way on how to use the role. Strange tho that it didn't work by changing the defaults.
Variables in the playbook should overwrite default variables. If nothing has been defined in playbook the default will be used.
